mlflow/models/model.py

```diff
@@ -104,6 +104,7 @@ def __init__(
         metadata: Optional[dict[str, Any]] = None,
         registered_model_version: Optional[int] = None,
         env_vars: Optional[list[str]] = None,
+        signature_from_type_hint: bool = False,
```

Comment:
nit: maybe `is_signature_from_type_hint`? at the moment it feels like this should contain the signature itself

```suggestion
        is_signature_from_type_hint: bool = False,
```